app/DAO/BakeryDatabase.scala

```diff
@@ -105,6 +105,47 @@ class BakeryDatabase @Inject() (
       .transact(bakeryTransactor.xa)
       .unsafeRunSync()
   }
+
+  def updateProduct(id: String, changesJson: JsValue): Int = {
+
+    /** Steps for updating a record
+      *  1. bring in the json, json should be the values to change:
+      *     - name, qty, price, name+qty, name+price, qty+price, name+qty+price
+      *  2. check what values are given to update, form query on these params
+      *  3. query with .update on the given id
+      *  4. run the transaction
+      */
+    val newName = (changesJson \ "name").asOpt[String]
+    val newQty = (changesJson \ "quantity").asOpt[Int]
+    val newPrice = (changesJson \ "price").asOpt[Double]
+    val newStamp = OffsetDateTime.now()
+    var query = sql""""""
+    if (newName.isDefined && newQty.isDefined && newPrice.isDefined) {
```

Comment:
what you are doing here is pattern matching specific items. It is generally easier to read (and notice if you are missing cases) if you use the pattern matching syntax:
```
(newName, newQty, newPrice) match {
    case (Some(name), Some(quantity), Some(price) => 
    case (Some(name), Some(quantity), None =>
    ... // etc
    case _ => // default
```